Add devices features

juanmbellini opened the pull request:
**Wait till #12 is merged**

Closes #7

- Add devices stuff models and migration script
- Add persistence support for devices
- Add devices support for devices
- Add support classes for web layer
- Add web support for devices
- Add authorization stuff

juanmbellini committed:
Add devices stuff models and migration script

- Add Device model
- Add DeviceRegistration model
- Add migration script to have model tables

julianrod94 commented on server-core/src/main/java/ar/edu/itba/iot/carne_iot/server/models/Device.java:
```diff
@@ -0,0 +1,146 @@
+     * The last temperature measured (and informed) by this device.
+     */
+    @Column(name = "temperature", precision = PRECISION, scale = SCALE)
+    private BigDecimal temperature;
```
hace falta big decimal? Con un float no alcanza?

julianrod94 replied:
@juanmbellini

juanmbellini replied:
Ya lo hablamos por telegram.

julianrod94 commented:
Salvo lo del BigInteger me cabe todo lo demás

juanmbellini commented:
Que es lo de BigInteger?

juanmbellini committed:
Make some changes in Device and DeviceRegistration models

- Include more behaviour to support more features

juanmbellini committed:
Refactor PersistenceHelper

- Make changes to have another validatePageable method, which receives a set of properties

juanmbellini committed:
Add persistence support for devices

juanmbellini committed:
Add service support for Devices

juanmbellini committed:
Add support classes to web layer

- Add Java8Instant json serializer (to transfer timestamps with the client)
- Add Base64url json serializer and deserializer to transfer Device ids in base64url encoding

juanmbellini committed:
Add DTOs for Devices

juanmbellini committed:
Fix UserServiceImpl to be make registration idempotent

- If the User already registered the device to its account, do not fail, but return without doing anything

juanmbellini committed:
Add endpoints for device management

- Add missing log messages and validations to UserEndpoint

juanmbellini committed:
Make base64url json serializer/deserializer use base64url helper

juanmbellini committed:
Extract isAdmin strategy to authorization helper class

- The isAdmin method will be used in several authorization methods

juanmbellini committed:
Add authorization stuff to devices operations

juanmbellini committed:
Add validations for state of entities

- Change state of Devices must be in a valid state of them

julianrod94 commented:
ahi lo vi todo.  Me parece que esta bien

juanmbellini committed:
Merge branch 'master' into feature/devices
